File: Nx3CS/bin/test/src/nx3/PColumnsDistancesCalculator.cs

#pragma warning disable 109, 114, 219, 429, 168, 162
namespace nx3{
	public  class PColumnsDistancesCalculator : global::haxe.lang.HxObject {
		public    PColumnsDistancesCalculator(global::haxe.lang.EmptyObject empty){
			unchecked {
				#line 14 "F:\\nx3\\src\\nx3\\PColumnsDistancesCalculator.hx"
				{
				}
				
			}
			#line default
		}
		
		
		public    PColumnsDistancesCalculator(global::nx3.PBar bar){
			unchecked {
				#line 18 "F:\\nx3\\src\\nx3\\PColumnsDistancesCalculator.hx"
				global::nx3.PColumnsDistancesCalculator.__hx_ctor_nx3_PColumnsDistancesCalculator(this, bar);
			}
			#line default
		}
		
		
		public static   void __hx_ctor_nx3_PColumnsDistancesCalculator(global::nx3.PColumnsDistancesCalculator __temp_me94, global::nx3.PBar bar){
			unchecked {
				#line 20 "F:\\nx3\\src\\nx3\\PColumnsDistancesCalculator.hx"
				__temp_me94.bar = bar;
				__temp_me94.prevLeftComplex = new global::haxe.ds.IntMap<object>();
			}
			#line default
		}
		
		
		public static  new object __hx_createEmpty(){
			unchecked {
				#line 14 "F:\\nx3\\src\\nx3\\PColumnsDistancesCalculator.hx"
				return new global::nx3.PColumnsDistancesCalculator(((global::haxe.lang.EmptyObject) (global::haxe.lang.EmptyObject.EMPTY) ));
			}
			#line default
		}
		
		
		public static  new object __hx_create(global::Array arr){
			unchecked {
				#line 14 "F:\\nx3\\src\\nx3\\PColumnsDistancesCalculator.hx"
				return new global::nx3.PColumnsDistancesCalculator(((global::nx3.PBar) (arr[0]) ));
			}
			#line default
		}
		
		
		public  global::nx3.PBar bar;
		
		public virtual   void calculate(){
			unchecked {
				#line 27 "F:\\nx3\\src\\nx3\\PColumnsDistancesCalculator.hx"
				global::nx3.PColumn leftColumn = default(global::nx3.PColumn);
				double xposition = 0.0;
				{
					#line 29 "F:\\nx3\\src\\nx3\\PColumnsDistancesCalculator.hx"
					int _g1 = 0;
					#line 29 "F:\\nx3\\src\\nx3\\PColumnsDistancesCalculator.hx"
					int _g = this.bar.getColumns().length;
					#line 29 "F:\\nx3\\src\\nx3\\PColumnsDistancesCalculator.hx"
					while (( _g1 < _g )){
						#line 29 "F:\\nx3\\src\\nx3\\PColumnsDistancesCalculator.hx"
						int columnIdx = _g1++;
						#line 31 "F:\\nx3\\src\\nx3\\PColumnsDistancesCalculator.hx"
						global::nx3.PColumn rightColumn = ((global::nx3.PColumn) (this.bar.getColumns()[columnIdx]) );
						rightColumn.mdistance = new global::haxe.lang.Null<double>(((double) (0) ), true);
						#line 35 "F:\\nx3\\src\\nx3\\PColumnsDistancesCalculator.hx"
						if (( columnIdx == 0 )) {
							#line 37 "F:\\nx3\\src\\nx3\\PColumnsDistancesCalculator.hx"
							int complexId = 0;
							{
								#line 38 "F:\\nx3\\src\\nx3\\PColumnsDistancesCalculator.hx"
								int _g2 = 0;
								#line 38 "F:\\nx3\\src\\nx3\\PColumnsDistancesCalculator.hx"
								global::nx3.PColumn __temp_stmt241 = default(global::nx3.PColumn);
								#line 38 "F:\\nx3\\src\\nx3\\PColumnsDistancesCalculator.hx"
								{
									#line 38 "F:\\nx3\\src\\nx3\\PColumnsDistancesCalculator.hx"
									global::Array<object> array = this.bar.getColumns();
									#line 38 "F:\\nx3\\src\\nx3\\PColumnsDistancesCalculator.hx"
									__temp_stmt241 = ((global::nx3.PColumn) (array[0]) );
								}
								
								#line 38 "F:\\nx3\\src\\nx3\\PColumnsDistancesCalculator.hx"
								global::Array<object> _g3 = __temp_stmt241.getComplexes();
								#line 38 "F:\\nx3\\src\\nx3\\PColumnsDistancesCalculator.hx"
								while (( _g2 < _g3.length )){
									#line 38 "F:\\nx3\\src\\nx3\\PColumnsDistancesCalculator.hx"
									global::nx3.PComplex complex = ((global::nx3.PComplex) (_g3[_g2]) );
									#line 38 "F:\\nx3\\src\\nx3\\PColumnsDistancesCalculator.hx"
									 ++ _g2;
									#line 40 "F:\\nx3\\src\\nx3\\PColumnsDistancesCalculator.hx"
									this.prevLeftComplex.@set(complexId, new global::haxe.lang.DynamicObject(new global::Array<int>(new int[]{50636009}), new global::Array<object>(new object[]{complex}), new global::Array<int>(new int[]{1050990343}), new global::Array<double>(new double[]{((double) (0) )})));
									complexId++;
								}
								
							}
							
						}
						 else {
							#line 45 "F:\\nx3\\src\\nx3\\PColumnsDistancesCalculator.hx"
							if (( columnIdx < this.bar.getColumns().length )) {
								#line 47 "F:\\nx3\\src\\nx3\\PColumnsDistancesCalculator.hx"
								global::Array<object> leftComplexes = leftColumn.getComplexes();
								global::Array<object> rightComplexes = rightColumn.getComplexes();
								#line 50 "F:\\nx3\\src\\nx3\\PColumnsDistancesCalculator.hx"
								double columndistance = 0.0;
								{
									#line 51 "F:\\nx3\\src\\nx3\\PColumnsDistancesCalculator.hx"
									int _g31 = 0;
									#line 51 "F:\\nx3\\src\\nx3\\PColumnsDistancesCalculator.hx"
									int _g21 = leftComplexes.length;
									#line 51 "F:\\nx3\\src\\nx3\\PColumnsDistancesCalculator.hx"
									while (( _g31 < _g21 )){
										#line 51 "F:\\nx3\\src\\nx3\\PColumnsDistancesCalculator.hx"
										int complexIdx = _g31++;
										#line 53 "F:\\nx3\\src\\nx3\\PColumnsDistancesCalculator.hx"
										global::nx3.PComplex leftComplex = ((global::nx3.PComplex) (leftComplexes[complexIdx]) );
										global::nx3.PComplex rightComplex = ((global::nx3.PComplex) (rightComplexes[complexIdx]) );
										double distance = this.getComplexDistances(columnIdx, complexIdx, leftComplex, rightComplex);
										#line 57 "F:\\nx3\\src\\nx3\\PColumnsDistancesCalculator.hx"
										columndistance = global::System.Math.Max(((double) (columndistance) ), ((double) (distance) ));
									}
									
								}
								
								#line 60 "F:\\nx3\\src\\nx3\\PColumnsDistancesCalculator.hx"
								{
									#line 60 "F:\\nx3\\src\\nx3\\PColumnsDistancesCalculator.hx"
									double number = columndistance;
									#line 60 "F:\\nx3\\src\\nx3\\PColumnsDistancesCalculator.hx"
									int precision = 6;
									#line 60 "F:\\nx3\\src\\nx3\\PColumnsDistancesCalculator.hx"
									number = ( number * global::System.Math.Pow(((double) (10) ), ((double) (precision) )) );
									#line 60 "F:\\nx3\\src\\nx3\\PColumnsDistancesCalculator.hx"
									number = ( global::Math.round(number) / global::System.Math.Pow(((double) (10) ), ((double) (precision) )) );
									#line 60 "F:\\nx3\\src\\nx3\\PColumnsDistancesCalculator.hx"
									columndistance = number;
								}
								
								#line 61 "F:\\nx3\\src\\nx3\\PColumnsDistancesCalculator.hx"
								leftColumn.mdistance = new global::haxe.lang.Null<double>(columndistance, true);
								#line 63 "F:\\nx3\\src\\nx3\\PColumnsDistancesCalculator.hx"
								xposition += columndistance;
								rightColumn.mposition = new global::haxe.lang.Null<double>(xposition, true);
							}
							
						}
						
						#line 68 "F:\\nx3\\src\\nx3\\PColumnsDistancesCalculator.hx"
						if (( columnIdx == ( this.bar.getColumns().length - 1 ) )) {
							#line 70 "F:\\nx3\\src\\nx3\\PColumnsDistancesCalculator.hx"
							global::nx3.PColumn lastColumn = ((global::nx3.PColumn) (this.bar.getColumns()[columnIdx]) );
							lastColumn.mdistance = new global::haxe.lang.Null<double>(lastColumn.getRightX(), true);
							return ;
						}
						
						#line 74 "F:\\nx3\\src\\nx3\\PColumnsDistancesCalculator.hx"
						leftColumn = rightColumn;
					}
					
				}
				
			}
			#line default
		}
		
		
		public  global::haxe.ds.IntMap<object> prevLeftComplex;
		
		public virtual   double getComplexDistances(int columnIdx, int complexIdx, global::nx3.PComplex leftComplex, global::nx3.PComplex rightComplex){
			unchecked {
				#line 84 "F:\\nx3\\src\\nx3\\PColumnsDistancesCalculator.hx"
				if (( rightComplex == default(global::nx3.PComplex) )) {
					#line 87 "F:\\nx3\\src\\nx3\\PColumnsDistancesCalculator.hx"
					if (( leftComplex != default(global::nx3.PComplex) )) {
						#line 89 "F:\\nx3\\src\\nx3\\PColumnsDistancesCalculator.hx"
						int leftColumnIdx = ( columnIdx - 1 );
						#line 91 "F:\\nx3\\src\\nx3\\PColumnsDistancesCalculator.hx"
						this.prevLeftComplex.@set(complexIdx, new global::haxe.lang.DynamicObject(new global::Array<int>(new int[]{50636009}), new global::Array<object>(new object[]{leftComplex}), new global::Array<int>(new int[]{1050990343}), new global::Array<double>(new double[]{((double) (leftColumnIdx) )})));
					}
					
					#line 94 "F:\\nx3\\src\\nx3\\PColumnsDistancesCalculator.hx"
					return ((double) (0) );
				}
				
				#line 97 "F:\\nx3\\src\\nx3\\PColumnsDistancesCalculator.hx"
				if (( leftComplex == default(global::nx3.PComplex) )) {
					#line 100 "F:\\nx3\\src\\nx3\\PColumnsDistancesCalculator.hx"
					int currentLeftColumIdx = ( columnIdx - 1 );
					int prevLeftColumnIdx = ((int) (global::haxe.lang.Runtime.getField_f(this.prevLeftComplex.@get(complexIdx).@value, "columnIdx", 1050990343, true)) );
					#line 103 "F:\\nx3\\src\\nx3\\PColumnsDistancesCalculator.hx"
					double currentLeftXPos = ((global::nx3.PColumn) (this.bar.getColumns()[currentLeftColumIdx]) ).getMPosition();
					double prevLeftXPos = ((global::nx3.PColumn) (this.bar.getColumns()[prevLeftColumnIdx]) ).getMPosition();
					#line 107 "F:\\nx3\\src\\nx3\\PColumnsDistancesCalculator.hx"
					double distanceBenefit = ( currentLeftXPos - prevLeftXPos );
					#line 109 "F:\\nx3\\src\\nx3\\PColumnsDistancesCalculator.hx"
					global::nx3.PComplex currentLeftComplex = ((global::nx3.PComplex) (global::haxe.lang.Runtime.getField(this.prevLeftComplex.@get(complexIdx).@value, "leftComplex", 50636009, true)) );
					double distance = new global::nx3.PComplexDistancesCalculator().getDistance(currentLeftComplex, rightComplex);
					#line 120 "F:\\nx3\\src\\nx3\\PColumnsDistancesCalculator.hx"
					double actualDistance = global::System.Math.Max(((double) (0) ), ((double) (( distance - distanceBenefit )) ));
					return actualDistance;
				}
				
				#line 124 "F:\\nx3\\src\\nx3\\PColumnsDistancesCalculator.hx"
				int leftColumnIdx1 = ( columnIdx - 1 );
				#line 126 "F:\\nx3\\src\\nx3\\PColumnsDistancesCalculator.hx"
				double distance1 = new global::nx3.PComplexDistancesCalculator().getDistance(leftComplex, rightComplex);
				#line 128 "F:\\nx3\\src\\nx3\\PColumnsDistancesCalculator.hx"
				this.prevLeftComplex.@set(complexIdx, new global::haxe.lang.DynamicObject(new global::Array<int>(new int[]{50636009}), new global::Array<object>(new object[]{leftComplex}), new global::Array<int>(new int[]{1050990343}), new global::Array<double>(new double[]{((double) (leftColumnIdx1) )})));
				#line 132 "F:\\nx3\\src\\nx3\\PColumnsDistancesCalculator.hx"
				return distance1;
			}
			#line default
		}
		
		
		public override   object __hx_setField(string field, int hash, object @value, bool handleProperties){
			unchecked {
				#line 14 "F:\\nx3\\src\\nx3\\PColumnsDistancesCalculator.hx"
				switch (hash){
					case 430591158:
					{
						#line 14 "F:\\nx3\\src\\nx3\\PColumnsDistancesCalculator.hx"
						this.prevLeftComplex = ((global::haxe.ds.IntMap<object>) (global::haxe.ds.IntMap<object>.__hx_cast<object>(((global::haxe.ds.IntMap) (@value) ))) );
						#line 14 "F:\\nx3\\src\\nx3\\PColumnsDistancesCalculator.hx"
						return @value;
					}
					
					
					case 4895187:
					{
						#line 14 "F:\\nx3\\src\\nx3\\PColumnsDistancesCalculator.hx"
						this.bar = ((global::nx3.PBar) (@value) );
						#line 14 "F:\\nx3\\src\\nx3\\PColumnsDistancesCalculator.hx"
						return @value;
					}
					
					
					default:
					{
						#line 14 "F:\\nx3\\src\\nx3\\PColumnsDistancesCalculator.hx"
						return base.__hx_setField(field, hash, @value, handleProperties);
					}
					
				}
				
			}
			#line default
		}
		
		
		public override   object __hx_getField(string field, int hash, bool throwErrors, bool isCheck, bool handleProperties){
			unchecked {
				#line 14 "F:\\nx3\\src\\nx3\\PColumnsDistancesCalculator.hx"
				switch (hash){
					case 713611268:
					{
						#line 14 "F:\\nx3\\src\\nx3\\PColumnsDistancesCalculator.hx"
						return ((global::haxe.lang.Function) (new global::haxe.lang.Closure(((object) (this) ), ((string) ("getComplexDistances") ), ((int) (713611268) ))) );
					}
					
					
					case 430591158:
					{
						#line 14 "F:\\nx3\\src\\nx3\\PColumnsDistancesCalculator.hx"
						return this.prevLeftComplex;
					}
					
					
					case 90871142:
					{
						#line 14 "F:\\nx3\\src\\nx3\\PColumnsDistancesCalculator.hx"
						return ((global::haxe.lang.Function) (new global::haxe.lang.Closure(((object) (this) ), ((string) ("calculate") ), ((int) (90871142) ))) );
					}
					
					
					case 4895187:
					{
						#line 14 "F:\\nx3\\src\\nx3\\PColumnsDistancesCalculator.hx"
						return this.bar;
					}
					
					
					default:
					{
						#line 14 "F:\\nx3\\src\\nx3\\PColumnsDistancesCalculator.hx"
						return base.__hx_getField(field, hash, throwErrors, isCheck, handleProperties);
					}
					
				}
				
			}
			#line default
		}
		
		
		public override   object __hx_invokeField(string field, int hash, global::Array dynargs){
			unchecked {
				#line 14 "F:\\nx3\\src\\nx3\\PColumnsDistancesCalculator.hx"
				switch (hash){
					case 713611268:
					{
						#line 14 "F:\\nx3\\src\\nx3\\PColumnsDistancesCalculator.hx"
						return this.getComplexDistances(((int) (global::haxe.lang.Runtime.toInt(dynargs[0])) ), ((int) (global::haxe.lang.Runtime.toInt(dynargs[1])) ), ((global::nx3.PComplex) (dynargs[2]) ), ((global::nx3.PComplex) (dynargs[3]) ));
					}
					
					
					case 90871142:
					{
						#line 14 "F:\\nx3\\src\\nx3\\PColumnsDistancesCalculator.hx"
						this.calculate();
						#line 14 "F:\\nx3\\src\\nx3\\PColumnsDistancesCalculator.hx"
						break;
					}
					
					
					default:
					{
						#line 14 "F:\\nx3\\src\\nx3\\PColumnsDistancesCalculator.hx"
						return base.__hx_invokeField(field, hash, dynargs);
					}
					
				}
				
				#line 14 "F:\\nx3\\src\\nx3\\PColumnsDistancesCalculator.hx"
				return default(object);
			}
			#line default
		}
		
		
		public override   void __hx_getFields(global::Array<object> baseArr){
			unchecked {
				#line 14 "F:\\nx3\\src\\nx3\\PColumnsDistancesCalculator.hx"
				baseArr.push("prevLeftComplex");
				#line 14 "F:\\nx3\\src\\nx3\\PColumnsDistancesCalculator.hx"
				baseArr.push("bar");
				#line 14 "F:\\nx3\\src\\nx3\\PColumnsDistancesCalculator.hx"
				{
					#line 14 "F:\\nx3\\src\\nx3\\PColumnsDistancesCalculator.hx"
					base.__hx_getFields(baseArr);
				}
				
			}
			#line default
		}
		
		
	}
}

File: Playground/TestDiff/java/src/nx3/PColumnsDistancesCalculator.java
package nx3;
import haxe.root.*;

@SuppressWarnings(value={"rawtypes", "unchecked"})
public  class PColumnsDistancesCalculator extends haxe.lang.HxObject
{
	public    PColumnsDistancesCalculator(haxe.lang.EmptyObject empty)
	{
		{
		}
		
	}
	
	
	public    PColumnsDistancesCalculator(nx3.PBar bar)
	{
		nx3.PColumnsDistancesCalculator.__hx_ctor_nx3_PColumnsDistancesCalculator(this, bar);
	}
	
	
	public static   void __hx_ctor_nx3_PColumnsDistancesCalculator(nx3.PColumnsDistancesCalculator __temp_me87, nx3.PBar bar)
	{
		__temp_me87.bar = bar;
		__temp_me87.prevLeftComplex = new haxe.ds.IntMap<java.lang.Object>();
	}
	
	
	public static   java.lang.Object __hx_createEmpty()
	{
		return new nx3.PColumnsDistancesCalculator(((haxe.lang.EmptyObject) (haxe.lang.EmptyObject.EMPTY) ));
	}
	
	
	public static   java.lang.Object __hx_create(haxe.root.Array arr)
	{
		return new nx3.PColumnsDistancesCalculator(((nx3.PBar) (arr.__get(0)) ));
	}
	
	
	public  nx3.PBar bar;
	
	public   void calculate()
	{
		nx3.PColumn leftColumn = null;
		double xposition = 0.0;
		{
			int _g1 = 0;
			int _g = this.bar.getColumns().length;
			while (( _g1 < _g ))
			{
				int columnIdx = _g1++;
				nx3.PColumn rightColumn = this.bar.getColumns().__get(columnIdx);
				rightColumn.mdistance = 0;
				if (( columnIdx == 0 )) 
				{
					int complexId = 0;
					{
						int _g2 = 0;
						nx3.PColumn __temp_stmt489 = null;
						{
							haxe.root.Array<nx3.PColumn> array = this.bar.getColumns();
							__temp_stmt489 = array.__get(0);
						}
						
						haxe.root.Array<nx3.PComplex> _g3 = __temp_stmt489.getComplexes();
						while (( _g2 < _g3.length ))
						{
							nx3.PComplex complex = _g3.__get(_g2);
							 ++ _g2;
							this.prevLeftComplex.set(complexId, new haxe.lang.DynamicObject(new haxe.root.Array<java.lang.String>(new java.lang.String[]{"leftComplex"}), new haxe.root.Array<java.lang.Object>(new java.lang.Object[]{complex}), new haxe.root.Array<java.lang.String>(new java.lang.String[]{"columnIdx"}), new haxe.root.Array<java.lang.Object>(new java.lang.Object[]{((java.lang.Object) (((double) (0) )) )})));
							complexId++;
						}
						
					}
					
				}
				 else 
				{
					if (( columnIdx < this.bar.getColumns().length )) 
					{
						haxe.root.Array<nx3.PComplex> leftComplexes = leftColumn.getComplexes();
						haxe.root.Array<nx3.PComplex> rightComplexes = rightColumn.getComplexes();
						double columndistance = 0.0;
						{
							int _g31 = 0;
							int _g21 = leftComplexes.length;
							while (( _g31 < _g21 ))
							{
								int complexIdx = _g31++;
								nx3.PComplex leftComplex = leftComplexes.__get(complexIdx);
								nx3.PComplex rightComplex = rightComplexes.__get(complexIdx);
								double distance = this.getComplexDistances(columnIdx, complexIdx, leftComplex, rightComplex);
								columndistance = java.lang.Math.max(columndistance, distance);
							}
							
						}
						
						{
							double number = columndistance;
							int precision = 6;
							number = ( number * java.lang.Math.pow(((double) (10) ), ((double) (precision) )) );
							number = ( ((int) (java.lang.Math.round(number)) ) / java.lang.Math.pow(((double) (10) ), ((double) (precision) )) );
							columndistance = number;
						}
						
						leftColumn.mdistance = columndistance;
						xposition += columndistance;
						rightColumn.mposition = xposition;
					}
					
				}
				
				if (( columnIdx == ( this.bar.getColumns().length - 1 ) )) 
				{
					nx3.PColumn lastColumn = this.bar.getColumns().__get(columnIdx);
					lastColumn.mdistance = lastColumn.getRightX();
					return ;
				}
				
				leftColumn = rightColumn;
			}
			
		}
		
	}
	
	
	public  haxe.ds.IntMap<java.lang.Object> prevLeftComplex;
	
	public   double getComplexDistances(int columnIdx, int complexIdx, nx3.PComplex leftComplex, nx3.PComplex rightComplex)
	{
		if (( rightComplex == null )) 
		{
			if (( leftComplex != null )) 
			{
				int leftColumnIdx = ( columnIdx - 1 );
				this.prevLeftComplex.set(complexIdx, new haxe.lang.DynamicObject(new haxe.root.Array<java.lang.String>(new java.lang.String[]{"leftComplex"}), new haxe.root.Array<java.lang.Object>(new java.lang.Object[]{leftComplex}), new haxe.root.Array<java.lang.String>(new java.lang.String[]{"columnIdx"}), new haxe.root.Array<java.lang.Object>(new java.lang.Object[]{((java.lang.Object) (((double) (leftColumnIdx) )) )})));
			}
			
			return ((double) (0) );
		}
		
		if (( leftComplex == null )) 
		{
			int currentLeftColumIdx = ( columnIdx - 1 );
			int prevLeftColumnIdx = ((int) (haxe.lang.Runtime.getField_f(this.prevLeftComplex.get(((int) (complexIdx) )), "columnIdx", true)) );
			double currentLeftXPos = this.bar.getColumns().__get(currentLeftColumIdx).getMPosition();
			double prevLeftXPos = this.bar.getColumns().__get(prevLeftColumnIdx).getMPosition();
			double distanceBenefit = ( currentLeftXPos - prevLeftXPos );
			nx3.PComplex currentLeftComplex = ((nx3.PComplex) (haxe.lang.Runtime.getField(this.prevLeftComplex.get(((int) (complexIdx) )), "leftComplex", true)) );
			double distance = new nx3.PComplexDistancesCalculator().getDistance(currentLeftComplex, rightComplex);
			double actualDistance = java.lang.Math.max(((double) (0) ), ( distance - distanceBenefit ));
			return actualDistance;
		}
		
		int leftColumnIdx1 = ( columnIdx - 1 );
		double distance1 = new nx3.PComplexDistancesCalculator().getDistance(leftComplex, rightComplex);
		this.prevLeftComplex.set(complexIdx, new haxe.lang.DynamicObject(new haxe.root.Array<java.lang.String>(new java.lang.String[]{"leftComplex"}), new haxe.root.Array<java.lang.Object>(new java.lang.Object[]{leftComplex}), new haxe.root.Array<java.lang.String>(new java.lang.String[]{"columnIdx"}), new haxe.root.Array<java.lang.Object>(new java.lang.Object[]{((java.lang.Object) (((double) (leftColumnIdx1) )) )})));
		return distance1;
	}
	
	
	@Override public   java.lang.Object __hx_setField(java.lang.String field, java.lang.Object value, boolean handleProperties)
	{
		{
			boolean __temp_executeDef490 = true;
			switch (field.hashCode())
			{
				case -530493834:
				{
					if (field.equals("prevLeftComplex")) 
					{
						__temp_executeDef490 = false;
						this.prevLeftComplex = ((haxe.ds.IntMap<java.lang.Object>) (value) );
						return value;
					}
					
					break;
				}
				
				
				case 97299:
				{
					if (field.equals("bar")) 
					{
						__temp_executeDef490 = false;
						this.bar = ((nx3.PBar) (value) );
						return value;
					}
					
					break;
				}
				
				
			}
			
			if (__temp_executeDef490) 
			{
				return super.__hx_setField(field, value, handleProperties);
			}
			 else 
			{
				throw null;
			}
			
		}
		
	}
	
	
	@Override public   java.lang.Object __hx_getField(java.lang.String field, boolean throwErrors, boolean isCheck, boolean handleProperties)
	{
		{
			boolean __temp_executeDef491 = true;
			switch (field.hashCode())
			{
				case 1781619652:
				{
					if (field.equals("getComplexDistances")) 
					{
						__temp_executeDef491 = false;
						return ((haxe.lang.Function) (new haxe.lang.Closure(((java.lang.Object) (this) ), haxe.lang.Runtime.toString("getComplexDistances"))) );
					}
					
					break;
				}
				
				
				case 97299:
				{
					if (field.equals("bar")) 
					{
						__temp_executeDef491 = false;
						return this.bar;
					}
					
					break;
				}
				
				
				case -530493834:
				{
					if (field.equals("prevLeftComplex")) 
					{
						__temp_executeDef491 = false;
						return this.prevLeftComplex;
					}
					
					break;
				}
				
				
				case -1283652762:
				{
					if (field.equals("calculate")) 
					{
						__temp_executeDef491 = false;
						return ((haxe.lang.Function) (new haxe.lang.Closure(((java.lang.Object) (this) ), haxe.lang.Runtime.toString("calculate"))) );
					}
					
					break;
				}
				
				
			}
			
			if (__temp_executeDef491) 
			{
				return super.__hx_getField(field, throwErrors, isCheck, handleProperties);
			}
			 else 
			{
				throw null;
			}
			
		}
		
	}
	
	
	@Override public   java.lang.Object __hx_invokeField(java.lang.String field, haxe.root.Array dynargs)
	{
		{
			boolean __temp_executeDef492 = true;
			switch (field.hashCode())
			{
				case 1781619652:
				{
					if (field.equals("getComplexDistances")) 
					{
						__temp_executeDef492 = false;
						return this.getComplexDistances(((int) (haxe.lang.Runtime.toInt(dynargs.__get(0))) ), ((int) (haxe.lang.Runtime.toInt(dynargs.__get(1))) ), ((nx3.PComplex) (dynargs.__get(2)) ), ((nx3.PComplex) (dynargs.__get(3)) ));
					}
					
					break;
				}
				
				
				case -1283652762:
				{
					if (field.equals("calculate")) 
					{
						__temp_executeDef492 = false;
						this.calculate();
					}
					
					break;
				}
				
				
			}
			
			if (__temp_executeDef492) 
			{
				return super.__hx_invokeField(field, dynargs);
			}
			
		}
		
		return null;
	}
	
	
	@Override public   void __hx_getFields(haxe.root.Array<java.lang.String> baseArr)
	{
		baseArr.push("prevLeftComplex");
		baseArr.push("bar");
		{
			super.__hx_getFields(baseArr);
		}
		
	}
	
	
}


